engine/access/ingestion/engine.go

```diff
@@ -337,9 +452,130 @@ func (e *Engine) processAvailableExecutionReceipts(ctx context.Context) error {
 		if err := e.handleExecutionReceipt(msg.OriginID, receipt); err != nil {
 			return err
 		}
+
+		// notify, to fetch and store transaction result error messages for block
+		e.txResultErrorMessagesChan <- receipt.BlockID
+	}
+}
+
+// processTransactionResultErrorMessagesByReceipts handles error messages related to transaction
+// results by reading from the error messages channel and processing them accordingly.
+//
+// This function listens for messages on the txResultErrorMessagesChan channel and
+// processes each transaction result error message as it arrives.
+//
+// No errors are expected during normal operation.
+func (e *Engine) processTransactionResultErrorMessagesByReceipts(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case blockID := <-e.txResultErrorMessagesChan:
+			err := e.handleTransactionResultErrorMessages(ctx, blockID)
+			if err != nil {
+				// if an error reaches this point, it is unexpected
+				ctx.Throw(err)
+				return
+			}
+		}
 	}
 }
 
+// handleTransactionResultErrorMessages processes transaction result error messages for a given block ID.
+// It retrieves error messages from the backend if they do not already exist in storage.
+//
+// The function first checks if error messages for the given block ID are already present in storage.
+// If they are not, it fetches the messages from execution nodes and stores them.
+//
+// Parameters:
+// - ctx: The context for managing cancellation and deadlines during the operation.
+// - blockID: The identifier of the block for which transaction result error messages need to be processed.
+//
+// No errors are expected during normal operation.
+func (e *Engine) handleTransactionResultErrorMessages(ctx context.Context, blockID flow.Identifier) error {
+	if e.transactionResultErrorMessages == nil {
+		return nil
+	}
+
+	exists, err := e.transactionResultErrorMessages.Exists(blockID)
+	if err != nil {
+		return fmt.Errorf("could not check existance of transaction result error messages: %w", err)
+	}
+
+	if exists {
+		return nil
+	}
+
+	// retrieves error messages from the backend if they do not already exist in storage
```

Comment:
can you add a debug level log event in this method that logs the messages are being downloaded

Reply:
[9ce9d22](https://github.com/onflow/flow-go/pull/6468/commits/9ce9d2212bb954a9c8a6e1e77d65c40227ae4e11)